projects/map/src/lib/map/map.component.ts

```diff
@@ -15,15 +18,51 @@ import { MapService } from '../map-service/map.service';
     }
   `],
 })
-export class MapComponent implements AfterViewInit {
+export class MapComponent implements AfterViewInit, OnDestroy {
+  private static BOOKMARK_DESCRIPTOR: BookmarkFragmentStringDescriptor = { type: 'string', identifier: '' };
+
+  private destroyed = new Subject();
 
   constructor(
     private el: ElementRef,
     private mapService: MapService,
-  ) { }
+    private bookmarkService: BookmarkService,
```

Comment:
It would be better to move this code to the `core` project so that the `map` project is not dependant on bookmarks. It would be better to use a service / component to listen to bookmarks and call the map-service API to zoom to locations / update the bookmark.